sources/lib/structures/typed.ml

```diff
@@ -52,6 +52,10 @@ type oplogic =
     OPand | OPor | OPxor | OPimp | OPnot | OPiff
   | OPif
 
+type pattern =
+  | Constr of { name : Hstring.t ; args : (Var.t * Hstring.t * Ty.t) list}
```

Comment:
Again, some documentation ?
Particularly, what is the semantic of the `Hstring.t` in each triple in the list of arguments ?